presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkPhysicalResourceCalculator.java

```diff
@@ -0,0 +1,86 @@
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.plan.PlanNode;
+import io.airlift.units.DataSize;
+
+import static com.facebook.presto.spark.PhysicalResourceSettings.DISABLED_PHYSICAL_RESOURCE_SETTING;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getAverageInputDataSizePerExecutor;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getAverageInputDataSizePerPartition;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getMaxExecutorCount;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getMaxHashPartitionCount;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getMinExecutorCount;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.getMinHashPartitionCount;
+import static com.facebook.presto.spark.PrestoSparkSessionProperties.isSparkResourceAllocationStrategyEnabled;
+import static io.airlift.units.DataSize.Unit.BYTE;
+
+public class PrestoSparkPhysicalResourceCalculator
+{
+    private static final Logger log = Logger.get(PrestoSparkPhysicalResourceCalculator.class);
+
+    public PhysicalResourceSettings calculate(PlanNode plan, Metadata metaData, Session session)
+    {
+        if (!isSparkResourceAllocationStrategyEnabled(session)) {
+            return new PhysicalResourceSettings(0, 0);
+        }
+
+        double inputDataInBytes = new PrestoSparkSourceStatsCollector(metaData, session).collectSourceStats(plan);
+
+        if (inputDataInBytes < 0) {
+            log.warn(String.format("Input data statistics missing, inputDataInBytes=%.2f skipping automatic resource tuning.", inputDataInBytes));
+            return DISABLED_PHYSICAL_RESOURCE_SETTING;
+        }
+
+        if ((inputDataInBytes > Double.MAX_VALUE) || (Double.isNaN(inputDataInBytes))) {
+            log.warn(String.format("Failed to retrieve correct size, data read=%.2f, skipping automatic resource tuning.", inputDataInBytes));
+            return DISABLED_PHYSICAL_RESOURCE_SETTING;
+        }
+
+        DataSize inputSize = new DataSize(inputDataInBytes, BYTE);
+        int executorCount = calculateExecutorCount(session, inputSize);
+        int hashPartitionCount = calculateHashPartitionCount(session, inputSize);
+
+        return new PhysicalResourceSettings(executorCount, hashPartitionCount);
+    }
+
+    private int calculateExecutorCount(Session session, DataSize inputData)
+    {
+        int minExecutorCount = getMinExecutorCount(session);
+        int maxExecutorCount = getMaxExecutorCount(session);
+
+        long averageInputDataSizePerExecutorInBytes = getAverageInputDataSizePerExecutor(session).toBytes();
+        long calculatedNumberOfExecutors = inputData.toBytes() / averageInputDataSizePerExecutorInBytes;
+
+        int updatedExecutorCount = Math.max((int) calculatedNumberOfExecutors, minExecutorCount);
+
+        return (Math.max(minExecutorCount, Math.min(maxExecutorCount, updatedExecutorCount)));
```

Comment:
Line 69 is redundant here. We can remove it. Also, no need for enclosing brackets for outer Math.max()